### PR TITLE
Hide stable releases by default

### PR DESCRIPTION
**Pull request by mkondratek**
## Test plan
- do the release 😅

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -377,15 +377,21 @@ tasks {
...
+
     // pluginVersion is based on the SemVer (https://semver.org) and supports pre-release labels,
     // like 2.1.7-nightly
     // Specify pre-release label to publish the plugin in a custom Release Channel automatically.
     // Read more:
     // https://plugins.jetbrains.com/docs/intellij/deployment.html#specifying-a-release-channel
-
-    channels.set(
-        listOf(
-            properties("pluginVersion").split('-').getOrElse(1) { "default" }.split('.').first()))
```
Comment (mkondratek) [inline build.gradle.kts:388]:
```
.split('.').first()
```

This ☝️ was redundant. Historically, the plugin supported versions like `2.1.7-alpha.3`. It's no longer a case for us.

File: build.gradle.kts [hunk 2]
```diff
@@ -377,15 +377,21 @@ tasks {
...
+    val channel = properties("pluginVersion").split('-').getOrElse(1) { "default" }
+    channels.set(listOf(channel))
+
+    if (channel == "default") {
+      // The published version WILL NOT be available right after the JetBrains approval.
+      // Instead, we control if and when we want to make it available.
+      // (Note: there is ~48h waiting time for JetBrains approval).
+      hidden.set(true)
```
Comment (mkondratek) [inline build.gradle.kts:393]: 🕵️‍♂️

Comment (mkondratek) [reply build.gradle.kts:393]: See docs for more:

https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin.html#tasks-publishplugin

File: build.gradle.kts [hunk 3]
```diff
@@ -377,15 +377,21 @@ tasks {
...
-    channels.set(
-        listOf(
-            properties("pluginVersion").split('-').getOrElse(1) { "default" }.split('.').first()))
+    val channel = properties("pluginVersion").split('-').getOrElse(1) { "default" }
+    channels.set(listOf(channel))
+
+    if (channel == "default") {
+      // The published version WILL NOT be available right after the JetBrains approval.
+      // Instead, we control if and when we want to make it available.
+      // (Note: there is ~48h waiting time for JetBrains approval).
```
Comment (mkondratek) [inline build.gradle.kts:392]: See docs on the approval process:
https://plugins.jetbrains.com/legal/approval-guidelines

**Comment by pkukielka** [comment]
LGTM

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**